scripts/install_third_party.py

```diff
@@ -165,56 +190,117 @@ def return_json(source_url):
     return json.loads(response)
 
 
+def test_manifest_syntax(dependency_type, dependency_dict):
+    """This checks syntax of the manifest.json dependencies.
+
+    Display warning message when there is an error and terminate the program.
+    Args:
+      dependency_type: dependency download format.
+      dependency_dict: manifest.json dependency dict
+    """
+    keys = dependency_dict.keys()
+    mandatory_keys = DOWNLOAD_FORMATS_TO_MANIFEST_KEYS[
+        dependency_type]['mandatory_keys']
+    # Optional keys requires atleast one member of the pair
```

Comment:
Just to check: "at least one" or "exactly one"? I think it's the latter, in which case both this docstring and the check below need to be changed.